spring-session/src/main/java/org/springframework/session/AuthenticationParser.java

```diff
@@ -0,0 +1,31 @@
+package org.springframework.session;
```

Comment:
I prefer not to expose this logic for reasons explained in #372 Please move this logic within the mongo package and make it package scope or private.